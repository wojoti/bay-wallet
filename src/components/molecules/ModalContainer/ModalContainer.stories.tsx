import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {ModalContainer} from 'components/molecules';

export default {
  title: 'Molecules/ModalContainer',
  component: ModalContainer,
  argTypes: {
    onPress: {action: 'onPress'},
  },
} as ComponentMeta<typeof ModalContainer>;
const Template: ComponentStory<typeof ModalContainer> = args => (
  <ModalContainer {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  header: 'Header',
  description: 'Modal description',
  buttonText: 'Button text',
};
