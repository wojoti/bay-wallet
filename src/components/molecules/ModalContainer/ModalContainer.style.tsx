import {StyleSheet} from 'react-native';
import {primary, secondary, successGreen} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    backgroundColor: '#0000004a',
  },
  spacer: {
    marginBottom: 20,
  },
  box: {
    backgroundColor: secondary,
    borderRadius: 8,
    borderColor: secondary,
    width: '90%',
    paddingHorizontal: 40,
    paddingTop: 50,
    paddingBottom: 30,
  },
  circle: {
    borderRadius: 45,
    borderWidth: 1,
    borderColor: successGreen,
    backgroundColor: successGreen,
    width: 'auto',
    padding: 15,
    marginTop: 30,
    marginBottom: 40,

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 20,
    },
    shadowOpacity: 0.18,
    shadowRadius: 6.0,

    elevation: 10,
  },
  button: {
    backgroundColor: primary,
    width: '100%',
    height: 50,
  },
  buttonText: {
    color: secondary,
  },
});
export default style;
