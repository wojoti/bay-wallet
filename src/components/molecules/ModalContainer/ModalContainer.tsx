import {Button, Container, Header, Img, ImgType, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {grey} from 'theme/colors';
import style from './ModalContainer.style';

export type ModalContainerProps = PropsWithChildren<{
  onPress: () => void;
  header: string;
  description: string;
  buttonText: string;
  testId?: string;
}>;

const ModalContainer = ({
  onPress,
  header,
  description,
  buttonText,
  testId,
}: ModalContainerProps) => {
  return (
    <Container
      flex={1}
      customStyle={style.wrapper}
      testId={testId}
      align="center"
      justify="center">
      <Container customStyle={style.box} align="center">
        <Container align="center" customStyle={style.spacer}>
          <Header testId="modalcontainer-header">{header}</Header>
        </Container>

        <Label testId="modalcontainer-label" color={grey}>
          {description}
        </Label>
        <Container customStyle={style.circle} justify="center" align="center">
          <Img
            testId="modalcontainer-img"
            src={ImgType.success}
            height={25}
            width={27}
            alt={''}
          />
        </Container>
        <Button
          customStyle={style.button}
          customFontStyle={style.buttonText}
          onPress={onPress}
          testId="modalcontainer-button">
          {buttonText}
        </Button>
      </Container>
    </Container>
  );
};

export default ModalContainer;
