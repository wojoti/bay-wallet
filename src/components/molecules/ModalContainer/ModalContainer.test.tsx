import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import ModalContainer, {ModalContainerProps} from './ModalContainer';

const mockOnPress = jest.fn();
const props: ModalContainerProps = {
  testId: 'test-modalcontainer-id',
  header: 'Header',
  description: 'Modal description',
  buttonText: 'Button text',
  onPress: mockOnPress,
};
test('should match snapshot', () => {
  const tree = TestRenderer.create(<ModalContainer {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render ModalContainer', () => {
  render(<ModalContainer {...props} />);
  const ModalContainerElement = screen.getByTestId('test-modalcontainer-id');
  expect(ModalContainerElement).toBeOnTheScreen();
});
test('should render ModalContainer - header', () => {
  render(<ModalContainer {...props} />);
  const ModalContainerElement = screen.getByTestId('test-modalcontainer-id');
  expect(ModalContainerElement).toBeOnTheScreen();
  const headerElement = within(ModalContainerElement).getByTestId(
    'modalcontainer-header',
  );
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('Header');
});
test('should render ModalContainer - label', () => {
  render(<ModalContainer {...props} />);
  const ModalContainerElement = screen.getByTestId('test-modalcontainer-id');
  expect(ModalContainerElement).toBeOnTheScreen();
  const labelElement = within(ModalContainerElement).getByTestId(
    'modalcontainer-label',
  );
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveTextContent('Modal description');
});
test('should render ModalContainer - img', () => {
  render(<ModalContainer {...props} />);
  const ModalContainerElement = screen.getByTestId('test-modalcontainer-id');
  expect(ModalContainerElement).toBeOnTheScreen();
  const imgElement = within(ModalContainerElement).getByTestId(
    'modalcontainer-img',
  );
  expect(imgElement).toBeOnTheScreen();
});
test('should render ModalContainer - button', () => {
  render(<ModalContainer {...props} />);
  const ModalContainerElement = screen.getByTestId('test-modalcontainer-id');
  expect(ModalContainerElement).toBeOnTheScreen();
  const btnElement = within(ModalContainerElement).getByTestId(
    'modalcontainer-button',
  );
  expect(btnElement).toBeOnTheScreen();
  const textElement = within(btnElement).getByTestId('button-text');
  expect(textElement).toBeOnTheScreen();
  expect(textElement).toHaveTextContent('Button text');
});

test('should render ModalContainer - button - handle onPress', () => {
  render(<ModalContainer {...props} />);
  const ModalContainerElement = screen.getByTestId('test-modalcontainer-id');
  expect(ModalContainerElement).toBeOnTheScreen();
  const btnElement = within(ModalContainerElement).getByTestId(
    'modalcontainer-button',
  );
  expect(btnElement).toBeOnTheScreen();
  expect(mockOnPress).toBeCalledTimes(0);
  fireEvent.press(btnElement);
  expect(mockOnPress).toBeCalledTimes(1);
});
