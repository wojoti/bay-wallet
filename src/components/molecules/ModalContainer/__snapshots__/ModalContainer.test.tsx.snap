// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      },
      {
        "backgroundColor": "#0000004a",
      },
    ]
  }
  testID="test-modalcontainer-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 8,
          "paddingBottom": 30,
          "paddingHorizontal": 40,
          "paddingTop": 50,
          "width": "90%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "marginBottom": 20,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Poppins-Bold",
            "fontSize": 24,
            "letterSpacing": undefined,
            "textAlign": "left",
          }
        }
        testID="modalcontainer-header"
      >
        Header
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#6B6B6B",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
      testID="modalcontainer-label"
    >
      Modal description
    </Text>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          },
          {
            "backgroundColor": "#3EB290",
            "borderColor": "#3EB290",
            "borderRadius": 45,
            "borderWidth": 1,
            "elevation": 10,
            "marginBottom": 40,
            "marginTop": 30,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 20,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 6,
            "width": "auto",
          },
        ]
      }
    >
      <SvgMock
        height={25}
        testID="modalcontainer-img"
        width={27}
      />
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 65,
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "backgroundColor": "#363062",
            "height": 50,
            "width": "100%",
          },
        ]
      }
      testID="modalcontainer-button"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Poppins-Bold",
              "fontSize": 18,
              "textAlign": "center",
            },
            {
              "color": "white",
            },
          ]
        }
        testID="button-text"
      >
        Button text
      </Text>
    </View>
  </View>
</View>
`;
